src/adapter/src/catalog.rs

```diff
@@ -2085,6 +2087,74 @@ impl<S: Append> Catalog<S> {
         Ok(())
     }
 
+    /// Create and insert the per replica log sources and log views.
+    ///
+    /// The replica with replica_id must already be present in the catalog state.
+    /// This function will allocate the oid's. The GlobalIds are already pre-allocated.
+    /// Views can depend on introspection sources and on other views.
+    fn insert_per_replica_items(
+        &self,
+        state: &mut CatalogState,
+        persisted_logs: &ConcreteComputeInstanceReplicaLogging,
+        replica_id: u64,
+    ) {
+        for (variant, source_id) in persisted_logs.get_logs() {
+            let oid = state.allocate_oid().expect("cannot return error here");
+            // TODO(lh): Once we get rid of legacy active logs, we should refactor the
+            // CatalogItem::Log. For now  we just use the log variant to lookup the unique CatalogItem
+            // in BUILTINS.
+            let log = BUILTINS::logs()
+                .find(|log| log.variant == variant)
+                .expect("variant must be included in builtins");
+
+            let source_name = QualifiedObjectName {
+                qualifiers: ObjectQualifiers {
+                    database_spec: ResolvedDatabaseSpecifier::Ambient,
+                    schema_spec: SchemaSpecifier::Id(self.get_mz_catalog_schema_id().clone()),
+                },
+                item: format!("{}_{}", log.name, replica_id),
+            };
+            state.insert_item(source_id, oid, source_name, CatalogItem::Log(log));
+        }
+
+        for (logview, id) in persisted_logs.get_views() {
+            let (sql_template, name_template) = logview.get_template();
+            assert!(sql_template.find("{}").is_some());
+            assert!(name_template.find("{}").is_some());
+            let name = name_template.replace("{}", &replica_id.to_string());
+            let sql = "CREATE VIEW mz_catalog.".to_string()
```

Comment:
There's a constant `MZ_CATALOG_SCHEMA` in `builtin.rs` that you can use for `"mz_catalog"`.